src/Common/Commands.Common/AzureDataCmdlet.cs

```diff
@@ -0,0 +1,143 @@
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using Microsoft.WindowsAzure.Commands.Common.Properties;
+
+namespace Microsoft.WindowsAzure.Commands.Common
+{
+    public class AzureDataCmdlet : AzurePSCmdlet
+    {
+        protected override Azure.Common.Authentication.Models.AzureContext DefaultContext
+        {
+            get
+            {
+                if (RMProfile != null && RMProfile.Context != null)
+                {
+                    return RMProfile.Context;
+                }
+
+                if (SMProfile == null || SMProfile.Context == null)
+                {
+                    throw new InvalidOperationException("There is no current context, please log in using Login-AzureRmAccount for Azure Resource Manager or Add-AzureAccount for Azure Service Management.");
+                }
+
+                return SMProfile.Context;
+            }
+        }
+
+        public AzureSMProfile SMProfile
+        {
+            get { return AzureSMProfileProvider.Instance.Profile; }
+        }
+
+        public AzureRMProfile RMProfile
+        {
+            get { return AzureRmProfileProvider.Instance.Profile; }
+        }
+
+        public static void ClearCurrentStorageAccount()
+        {
+            var RMProfile = AzureRmProfileProvider.Instance.Profile;
+            if (RMProfile != null && RMProfile.Context != null && 
+                RMProfile.Context.Subscription != null && RMProfile.Context.Subscription.IsPropertySet(AzureSubscription.Property.StorageAccount))
+            {
+                RMProfile.Context.Subscription.SetProperty(AzureSubscription.Property.StorageAccount, null);
+            }
+
+            var SMProfile = AzureSMProfileProvider.Instance.Profile;
+            if (SMProfile != null && SMProfile.Context != null && SMProfile.Context.Subscription != null && SMProfile.Context.Subscription.IsPropertySet(AzureSubscription.Property.StorageAccount))
+            {
+                SMProfile.Context.Subscription.SetProperty(AzureSubscription.Property.StorageAccount, null);
+            }
+        }
+
+        protected override void SaveDataCollectionProfile()
+        {
+             if (_dataCollectionProfile == null)
+            {
+                InitializeDataCollectionProfile();
+            }
+
+            string fileFullPath = Path.Combine(AzureSession.ProfileDirectory, AzurePSDataCollectionProfile.DefaultFileName);
+            var contents = JsonConvert.SerializeObject(_dataCollectionProfile);
+            if (!AzureSession.DataStore.DirectoryExists(AzureSession.ProfileDirectory))
+            {
+                AzureSession.DataStore.CreateDirectory(AzureSession.ProfileDirectory);
+            }
+            AzureSession.DataStore.WriteFile(fileFullPath, contents);
+            WriteWarning(string.Format(Resources.DataCollectionSaveFileInformation, fileFullPath));
+       }
+
+        protected override void PromptForDataCollectionProfileIfNotExists()
```

Comment:
Is this code duplicated between DataPlane and Management? Can we consolidate/unify it if thats the case?